Kernel/Net/LocalSocket.cpp

```diff
@@ -399,14 +399,23 @@ ErrorOr<void> LocalSocket::getsockopt(OpenFileDescription& description, int leve
 
     MutexLocker locker(mutex());
 
+    int buffer_size;
     socklen_t size;
     TRY(copy_from_user(&size, value_size.unsafe_userspace_ptr()));
 
     switch (option) {
     case SO_SNDBUF:
-        return ENOTSUP;
+        buffer_size = sizeof(*m_for_server.leak_ptr());
```

Comment:
This line is wrong on multiple levels, but most importantly: the size of the DoubleBuffer structure does not represent the buffer size, since the buffer is dynamically allocated. Please read up on the relevant APIs to find out how to actually get the buffer size.